tools/hardware/config.py

```diff
@@ -35,14 +36,15 @@ def get_device_page_bits(self) -> int:
         ''' Get page size in bits for mapping devices for this arch '''
         return self.get_page_bits()
 
-    def align_memory(self, regions: Set[Region]) -> List[Region]:
+    def align_memory(self, regions: Set[Region]) -> (List[Region], Set[Region], int):
```

Comment:
It seem the proper way to annotate do this is `Tuple[List[Region], Set[Region], int]`. So we should use this to avoid building legacy.